### PR TITLE
Updated README.md with Xcode version bump

### PR DESCRIPTION
**Pull request by stechiu**
### Summary of changes

- Changed Xcode version requirements in README file as per Apple guidelines https://developer.apple.com/news/?id=khzvxn8a

### Checklist

- [ ] Added a changelog entry

- @stechiu

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Welcome to Braintree's Drop-in SDK for iOS!
 
 ![Drop-in light theme](Images/client-sdk-ios-series-light.png "Drop-in light theme")
 
-**The Braintree iOS Drop-in SDK permits a deployment target of iOS 12.0 or higher.** It requires Xcode 14.1+ and Swift 5.7.1+.
+**The Braintree iOS Drop-in SDK permits a deployment target of iOS 12.0 or higher.** It requires Xcode 15.1+ and Swift 5.9+.
```
Comment (jaxdesmarais) [inline README.md:12]:
```suggestion
**The Braintree iOS Drop-in SDK permits a deployment target of iOS 12.0 or higher.** It requires Xcode 15.0+ and Swift 5.9+.
```

File: README.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ Welcome to Braintree's Drop-in SDK for iOS!
 We recommend using [Swift Package Manager](https://swift.org/package-manager/) or [CocoaPods](https://github.com/CocoaPods/CocoaPods) to integrate the Braintree Drop-in SDK with your project.
 
 ### Swift Package Manager
-_This feature is only available in v9. v9.Y.0 requires Xcode 12.5+._
+_This feature is only available in v9. v9.Y.0 requires Xcode 15.1+._
```
Comment (jaxdesmarais) [inline README.md:29]:
```suggestion
_This feature is only available in v9. v9.Y.0 requires Xcode 15.0+._
```

File: README.md [hunk 3]
```diff
@@ -45,7 +45,7 @@ Then run `pod install`.
 
 See our [`Podspec`](https://github.com/braintree/braintree-ios-drop-in/blob/main/BraintreeDropIn.podspec) for more information.
 
-*Note:* If you are using version 8.x.x of the Braintree Drop-in iOS SDK in Xcode 12, you may see the warning `The iOS Simulator deployment target is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99`. This will not prevent your app from compiling. This is a [CocoaPods issue](https://github.com/CocoaPods/CocoaPods/issues/7314) with a known workaround.
+*Note:* If you are using version 8.x.x of the Braintree Drop-in iOS SDK in Xcode 12, you may see the warning `The iOS Simulator deployment target is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.1`. This will not prevent your app from compiling. This is a [CocoaPods issue](https://github.com/CocoaPods/CocoaPods/issues/7314) with a known workaround.
```
Comment (jaxdesmarais) [inline README.md:48]: This is related to the iOS version not Xcode so we can revert this change.

Comment (stechiu) [reply README.md:48]: I overlooked the iOS portion, will revert. Thanks!

**Review by jaxdesmarais — APPROVED** [review]
SPM (this PR is from a fork) and UI test failure (has been failing recently) are expected.

**Review by KunJeongPark — APPROVED** [review]
